crates/optimism/evm/src/lib.rs

```diff
@@ -114,20 +115,33 @@ impl ConfigureEvm for OptimismEvmConfig {
     type DefaultExternalContext<'a> = ();
 
     fn evm<DB: Database>(&self, db: DB) -> Evm<'_, Self::DefaultExternalContext<'_>, DB> {
-        EvmBuilder::default().with_db(db).optimism().build()
+        // Define a default EVM configuration
+        let mut evm = {
+            self.default_external_context();
+            RethEvmBuilder::new(db, ())
+        }
+        .build();
+
+        // Apply the Optimism configuration to the EVM
+        evm.handler = Handler::optimism_with_spec(evm.handler.cfg.spec_id);
+        evm
     }
 
     fn evm_with_inspector<DB, I>(&self, db: DB, inspector: I) -> Evm<'_, I, DB>
     where
         DB: Database,
         I: GetInspector<DB>,
     {
-        EvmBuilder::default()
-            .with_db(db)
-            .with_external_context(inspector)
-            .optimism()
-            .append_handler_register(inspector_handle_register)
-            .build()
+        // Define a default EVM configuration with an inspector
+        let mut evm = {
+            self.default_external_context();
+            RethEvmBuilder::new(db, ())
+        }
+        .build_with_inspector(inspector);
+
+        // Apply the Optimism configuration to the EVM
+        evm.handler = Handler::optimism_with_spec(evm.handler.cfg.spec_id);
```

Comment:
is this correct?

isn't the inspector installed as a callback in the handler, so if we overwrite it like this, we lose it?

Reply:
> My idea here was to do everything to completely remove the REVM builder from the codebase.

ideally, yes but I really don't know if this is correct because from my understanding this removes the inspector